test/Cron.test.ts

```diff
@@ -0,0 +1,399 @@
+
+    it('emits an event', async () => {
+      const encodedSpec = await cron.cronStringToEncodedSpec('0 * * * *')
+      await expect(
+        cron.createCronJobFromEncodedSpec(
+          cronReceiver1.address,
+          handler1Sig,
+          encodedSpec,
+        ),
+      ).to.emit(cron, 'CronJobCreated')
+    })
+
+    it('is only callable by the owner', async () => {
+      const encodedSpec = await cron.cronStringToEncodedSpec('0 * * * *')
+      await expect(
+        cron
+          .connect(stranger)
+          .createCronJobFromEncodedSpec(
+            cronReceiver1.address,
+            handler1Sig,
+            encodedSpec,
+          ),
+      ).to.be.revertedWith(OWNABLE_ERR)
+    })
+  })
+
+  describe('deleteCronJob()', () => {
+    it("deletes a jobs by it's ID", async () => {
+      const encodedSpec = await cron.cronStringToEncodedSpec('0 * * * *')
+      await cron.createCronJobFromEncodedSpec(
+        cronReceiver1.address,
+        handler1Sig,
+        encodedSpec,
+      )
+      await cron.createCronJobFromEncodedSpec(
+        cronReceiver1.address,
+        handler2Sig,
+        encodedSpec,
+      )
+      await cron.createCronJobFromEncodedSpec(
+        cronReceiver2.address,
+        handler1Sig,
+        encodedSpec,
+      )
+      await cron.createCronJobFromEncodedSpec(
+        cronReceiver2.address,
+        handler2Sig,
+        encodedSpec,
+      )
+      await assertJobIDsEqual([1, 2, 3, 4])
+      await cron.deleteCronJob(2)
+      assert.equal(
+        (await cron.getCronJob(2)).target,
+        ethers.constants.AddressZero,
+      )
+      await assertJobIDsEqual([1, 3, 4])
+      await cron.deleteCronJob(1)
+      await assertJobIDsEqual([3, 4])
+      await cron.deleteCronJob(4)
+      await assertJobIDsEqual([3])
+      await cron.deleteCronJob(3)
+      await assertJobIDsEqual([])
+    })
+
+    it('emits an event', async () => {
+      await cron.createCronJobFromString(
+        cronReceiver1.address,
+        handler1Sig,
+        '0 0 31 * *',
+      )
+      await expect(cron.deleteCronJob(1)).to.emit(cron, 'CronJobDeleted')
+    })
+
+    it('is only callable by the owner', async () => {
+      await cron.createCronJobFromString(
+        cronReceiver1.address,
+        handler1Sig,
+        '0 0 31 * *',
+      )
+      await expect(cron.connect(stranger).deleteCronJob(1)).to.be.revertedWith(
+        OWNABLE_ERR,
+      )
+    })
+  })
+
+  describe('pause() / unpause()', () => {
+    it('is only callable by the owner', async () => {
+      await expect(cron.connect(stranger).pause()).to.be.reverted
+      await expect(cron.connect(stranger).unpause()).to.be.reverted
+    })
+
+    it('pauses / unpauses the contract', async () => {
+      expect(await cron.paused()).to.be.false
+      await cron.pause()
+      expect(await cron.paused()).to.be.true
+      await cron.unpause()
+      expect(await cron.paused()).to.be.false
+    })
+  })
+})
+
+// only run during yarn test:gas
```

Comment:
How is that enforced?

Reply:
package.json
```json
  "scripts": {
    "test": "hardhat test",
    "test:gas": "hardhat --config hardhat.gasUsage.config.ts test"
  }
 ```

Reply:
there's a filter on which mocha tests to run (it's a bit hacky, I'll admit)

Reply:
hardhat.config.ts
```ts
  mocha: {
    // skip gas tests, run those with yarn test:gas
    grep: '/Gas Usage/',
    invert: true,
    // ...
  }
```